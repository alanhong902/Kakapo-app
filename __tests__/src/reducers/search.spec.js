import { prop } from 'ramda';
import configureStore from 'stores/configureStore';
import { searchActions } from 'actions/';
import { kakapoRes } from '../helper';

const store = configureStore();

const currState = () => prop('search', store.getState());

test('[reducer/search] request loading', () => {
  store.dispatch(searchActions.request());
  expect(currState()).toMatchSnapshot();
});

test('[reducer/search] request success', () => {
  store.dispatch(searchActions.requestSuccess(kakapoRes, 'kakapofavs'));
  expect(currState()).toMatchSnapshot();
});
