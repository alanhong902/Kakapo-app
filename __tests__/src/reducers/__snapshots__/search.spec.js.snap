// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[reducer/search] request loading 1`] = `
Object {
  "kakapofavs": Array [],
  "loading": true,
  "soundcloud": Array [],
  "youtube": Array [],
}
`;

exports[`[reducer/search] request success 1`] = `
Object {
  "kakapofavs": Array [
    Object {
      "editing": false,
      "file": "birds",
      "img": "birds",
      "name": "Birds",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "animals, forest, outdoors, happy",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "bonfire",
      "img": "bonfire",
      "name": "Bonfire",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "fire, outdoors, camping, cosy",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "rain",
      "img": "rain",
      "name": "Rain",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "weather, water, outdoors",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "paris_cafe",
      "img": "cafe",
      "name": "Paris Cafe",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "people, crowd, city, coffee, chatter",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "cat_purr",
      "img": "cat",
      "name": "Cat Purring",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "pets, cat, indoor, content",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "chimes",
      "img": "chimes",
      "name": "Chimes",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "charm, outdoors, wind",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "fan",
      "img": "fan",
      "name": "Fan",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "machine, humming, indoors, repetitive",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "leaves",
      "img": "leaves",
      "name": "Leaves",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "weather, nature, windy, outdoors",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "paris_street_corner",
      "img": "paris",
      "name": "Paris Street Corner",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "people, city, cars, streets, outdoors",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "stream",
      "img": "water",
      "name": "Stream",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "weather, nature, water, running, outdoors",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "thunder",
      "img": "thunder",
      "name": "Thunder",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "storm, weather, nature, open, outdoors",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "train",
      "img": "train",
      "name": "Train",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "machine, travel, repetitive, soothing",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "waves",
      "img": "waves",
      "name": "Waves",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "ocean, nature, open, water",
      "volume": 0.5,
    },
    Object {
      "editing": false,
      "file": "wind",
      "img": "wind",
      "name": "Wind",
      "playing": false,
      "progress": 1,
      "recentlyDownloaded": false,
      "source": "file",
      "tags": "weather, nature, open, outdoors",
      "volume": 0.5,
    },
  ],
  "loading": false,
  "soundcloud": Array [],
  "youtube": Array [],
}
`;
